pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -199,9 +199,13 @@ func (p *LaunchTemplateProvider) blockDeviceMappings(blockDeviceMappings []*v1al
 				Iops:                blockDeviceMapping.EBS.IOPS,
 				Throughput:          blockDeviceMapping.EBS.Throughput,
 				KmsKeyId:            blockDeviceMapping.EBS.KMSKeyID,
-				VolumeSize:          aws.Int64(blockDeviceMapping.EBS.VolumeSize.ScaledValue(resource.Giga)),
```

Comment:
Optional: A getOrDefault() idiom or volumeSize would allow you to keep everything inlined and avoid naming decisions like blockDeviceMappingsRequest vs bdmr.